contracts/modules/Experimental/STO/POLYPeggedSTO.sol

```diff
@@ -0,0 +1,572 @@
+
+    ////////////
+    // Events //
+    ////////////
+
+    event SetAllowBeneficialInvestments(bool _allowed);
+    event SetNonAccreditedLimit(address _investor, uint256 _limit);
+    event SetAccredited(address _investor, bool _accredited);
+   /**
+    * Event for token purchase logging
+    * @param _purchaser who paid for the tokens
+    * @param _beneficiary who got the tokens
+    * @param _value amount paid for purchase in Fund Raise Type
+    * @param _valueUSD value of POLY invested in USD at the time of purchase
+    * @param _amount amount of tokens purchased
+    */
+    event TokenPurchase(
+        address indexed _purchaser,
+        address indexed _beneficiary,
+        uint256 _value,
+        uint256 _valueUSD,
+        uint256 _amount
+    );
+    event FundsReceived(
+        address indexed _purchaser,
+        address indexed _beneficiary,
+        FundRaiseType _fundRaiseType,
+        uint256 _receivedValue,
+        uint256 _spentValue
+    );
+    event ReserveTokenMint(address indexed _owner, address indexed _treasuryWallet, uint256 _tokens);
+    event SetWallet(address _oldWallet, address _newWallet);
+    event SetTreasuryWallet(address _oldWallet, address _newWallet);
+    event SetLimits(uint256 _minimumInvestment, uint256 _nonAccreditedLimit, uint256 _maxNonAccreditedInvestors);
+    event SetTimes(uint256 _startTime, uint256 _endTime);
+    event SetRate(uint256 _rate);
+    event SetCap(uint256 _cap);
+
+    ///////////////
+    // Modifiers //
+    ///////////////
+
+    modifier notStarted() {
+        /*solium-disable-next-line security/no-block-members*/
+        require(now < startTime, "STO already started");
+        _;
+    }
+
+    ///////////////////////
+    // STO Configuration //
+    ///////////////////////
+
+    constructor (address _securityToken, address _polyAddress) public
+    Module(_securityToken, _polyAddress)
+    {
+    }
+
+    /*
+     * @notice Function used to intialize the contract variables
+     * @param _startTime Unix timestamp at which offering get started
+     * @param _endTime Unix timestamp at which offering get ended
+     * @param _cap Maximum No. of token base units for sale
+     * @param _rate Token units a buyer gets multiplied by 10^18 per USD value of POLY
+     * @param _minimumInvestment Minimun investment in USD (* 10**18)
+     * @param _nonAccreditedLimit Default Limit in USD (* 10**18) for non-accredited investors
+     * @param _maxNonAccreditedInvestors Maximum number of non-accredited investors allowed (0 = unlimited)
+     * @param _wallet Ethereum account address to hold the funds
+     * @param _treasuryWallet Ethereum account where unsold Tokens will be sent
+     */
+    function configure(
+        uint256 _startTime,
+        uint256 _endTime,
+        uint256 _cap,
+        uint256 _rate,
+        uint256 _minimumInvestment,
+        uint256 _nonAccreditedLimit,
+        uint256 _maxNonAccreditedInvestors,
+        address payable _wallet,
```

Comment:
no need of payable here, as _wallet no longer receive ETHs

Reply:
As wallet is defined as payable in STOstorage.sol if I remove payable I get an error when compiling. I have left it as payable so it is using a consistent variable across STO's but could change it to a new variable if preferred. 
"Type address is not implicitly convertible to expected type address payable.                                                                                                                                                                                                                              wallet = _wallet;
Compilation failed. See above."

Reply:
Yeah, it makes sense it will not allow you to do that if it is already declared. No need to declare a new variable you can keep as it is. 
FYI - payable address work as non-payable address but the reverse is not true. To work it around you could use `address payable add = uint160(non-payable address)`.

Reply:
I googled this and got it to work using `wallet = address(uint160(_wallet));` Do you want me to make this change? While `wallet` won't receive ETH is this more or less clear to declare `_wallet `as payable than casting the non payable address to a payable one later?

Reply:
To receive the tokens minimum requirement is non-payable address. and it is possible to convert the non-payable address to payable as shown in the above comments.  As per your code, you are using payable wallet address because it is already declared in the `STOStorage.sol`. So I would prefer to keep as it is no need to convert it as it doesn't make sense to convert the payable address to non-payable because it will work same when you play with tokens.